trieste/space.py

```diff
@@ -246,17 +446,31 @@ class Box(SearchSpace):
     """
 
     @overload
-    def __init__(self, lower: Sequence[float], upper: Sequence[float]):
+    def __init__(
+        self,
+        lower: Sequence[float],
+        upper: Sequence[float],
+        constraints: Optional[Sequence[LinearConstraint | NonlinearConstraint]] = None,
+        ctol: float = 1e-7,
```

Comment:
default tolerance should depend on dtype I think

Reply:
I have updated the type to optionally include `TensorType`:
https://github.com/secondmind-labs/trieste/blob/f5a33b42342e41e464960972c3deffcccbbbeeda/trieste/space.py#L464

However, I am not sure what I should do with the value `dtype`. I copied the style from `ExpectedConstrainedImprovement` for default `min_feasibility_probability`:
https://github.com/secondmind-labs/trieste/blob/30a17e7fe4099e8ecd4aead386d787d8136d11f2/trieste/acquisition/function/function.py#L483